.github/workflows/test.yml

```diff
@@ -0,0 +1,107 @@
+name: Test
+on:
+  push:
+    branches:
+      - master
+      - release/**
+  pull_request:
+permissions:
+  contents: read
+defaults:
+  run:
+    shell: bash
+concurrency:
+  group: test-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  test:
+    name: Module Mode
+    runs-on: ${{ matrix.os }}-latest
+    env:
+      GO111MODULE: "on"
+      GOFLAGS: "-mod=readonly"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          # In order:
+          # * Module download cache
+          # * Build cache (Linux)
+          # * Build cache (Mac)
+          # * Build cache (Windows)
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ~/Library/Caches/go-build
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ matrix.go }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-${{ matrix.go }}-
+            ${{ runner.os }}-go-
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+      - name: Build
+        run: go build ./...
+      - name: Vet
+        run: go vet ./...
+      - name: Test
+        run: go test ./...
+      - name: Test (race)
+        run: go test -race ./...
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        go: ["1.16", "1.15", "1.14"]
+        os: [ubuntu, windows, macos]
+      fail-fast: false
+  test-gopath:
+    name: GOPATH Mode
+    runs-on: ubuntu-latest
+    env:
+      # We use two paths in GOPATH.
+      # The first one is where 'go get' will download dependencies, and
+      # the second is where sentry-go itself will be checked out.
+      GOPATH: ${{ github.workspace }}/deps:${{ github.workspace }}/main
+      GO111MODULE: "off"
+      WORKDIR: ${{ github.workspace }}/main/src/github.com/getsentry/sentry-go
+    defaults:
+      run:
+        working-directory: ${{ env.WORKDIR }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ env.WORKDIR }}
+      # TODO: cache dependencies
```

Comment:
Didn't enable this step for now because it requires a bit more thought on the cache key or how to update dependencies.

In GOPATH Mode we download all dependencies in their latest commits. With too aggressive caching or missing an update step (a la `go get -u`)  we'd be testing against stale dependencies.

We can't use `go get -d -t -u -v ./...` because that would also try to update sentry-go itself. The solution probably requires resorting to `go list` and would need to work no matter if we got a cache hit or miss. Deferring that for later.

This job is obviously prone to breakage (dependencies are changing out of our control) and not going to be required to merge changes.